set DEBUG_IMAGE=true when building debug docker image (#3250)
apollographql/router: .changesets/fix_garypen_3249_docker_heaptrack.md

> ### set DEBUG_IMAGE=true when building debug docker image ([Issue #3249](https://github.com/apollographql/router/issues/3249))

Reviewer: I think it would be more valuable to make this (not just here, but in all changesets) a user-impacting title statement whenever possible.  Otherwise, I don't know if this title means enough to the user as it stands.  It also makes it very digestible if we remove the bodies and just keep the titles.

```suggestion
### Restore missing debug tools in "debug" Docker images ([Issue #3249](https://github.com/apollographql/router/issues/3249))
```

> Router debug docker images were designed to make use of heaptrack for debugging memory issues. However, this functionality was broken when we changed to multi-architecture docker image builds.

Reviewer: Capitalization of other product names, mostly.

```suggestion
Debug Docker images were designed to make use of `heaptrack` for debugging memory issues. However, this functionality was inadvertently removed when we changed to multi-architecture Docker image builds.
```

This restores the heaptrack functionality to our debug docker images.

> fixes: #3249
> 

Reviewer: ```suggestion
```
